File: System/Protocol/Allors.Protocol.Json/Api/Pull/PullRequest.cs
// <copyright file="PullRequest.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Json.Api.Pull
{
    using System.Text.Json.Serialization;
    using Json;
    using Data;

    public class PullRequest
    {
        [JsonPropertyName("l")]
        public Pull[] List { get; set; }

        [JsonPropertyName("p")]
        public Procedure Procedure { get; set; }
    }
}

File: typescript/modules/libs/protocol/json/system/src/lib/api/pull/pull-request.ts
import { Procedure } from '../../data/procedure';
import { Pull } from '../../data/pull';
import { PullDependency } from './pull-dependency';

export interface PullRequest {
  /** Dependencies */
  d?: PullDependency[];

  /** List of Pulls */
  l?: Pull[];

  /** Procedure */
  p?: Procedure;
}
